streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1081,8 +1081,9 @@ private void shutdown(final boolean cleanRun) {
         streamsMetrics.removeAllSensors();
     }
 
+    // visible for testing
     @SuppressWarnings("ThrowableNotThrown")
-    private void shutdownTasksAndState(final boolean cleanRun) {
+    void shutdownTasksAndState(final boolean cleanRun) {
```

Comment:
@dguy If we need to access an inner function for lots of unit tests it usually indicates that our class design patterns are not good since unit test should be testing a class's out-facing behavior only; in other words cases that you need to trigger an inner function in unit tests should be rare, and I think reflection is fine as long as it is rarely used.